main/php_glob.h

```diff
@@ -0,0 +1,184 @@
+#ifdef GLOB_NOMATCH
+#define PHP_GLOB_NOMATCH GLOB_NOMATCH
+#endif
+#ifdef GLOB_NOSYS
+#define PHP_GLOB_NOSYS GLOB_NOSYS
+#endif
+#ifdef GLOB_ALTDIRFUNC
+#define PHP_GLOB_ALTDIRFUNC GLOB_ALTDIRFUNC
+#endif
+#ifdef GLOB_BRACE
+#define PHP_GLOB_BRACE GLOB_BRACE
+#endif
+#ifdef GLOB_MAGCHAR
+#define PHP_GLOB_MAGCHAR GLOB_MAGCHAR
+#endif
+#ifdef GLOB_NOMAGIC
+#define PHP_GLOB_NOMAGIC GLOB_NOMAGIC
+#endif
+#ifdef GLOB_QUOTE
+#define PHP_GLOB_QUOTE GLOB_QUOTE
+#endif
+#ifdef GLOB_TILDE
+#define PHP_GLOB_TILDE GLOB_TILDE
+#endif
+#ifdef GLOB_LIMIT
+#define PHP_GLOB_LIMIT GLOB_LIMIT
+#endif
+#ifdef GLOB_KEEPSTAT
+#define PHP_GLOB_KEEPSTAT GLOB_KEEPSTAT
+#endif
+
+#define php_glob_t glob_t
+#define php_glob glob
+#define php_globfree globfree
+#else
+
+#include "php.h"
+
+#ifndef PHP_WIN32
+# include <sys/cdefs.h>
+#endif
+
+#include "Zend/zend_stream.h"
+
+typedef struct {
+	size_t gl_pathc;	/* Count of total paths so far. */
+	size_t gl_matchc;	/* Count of paths matching pattern. */
+	size_t gl_offs;		/* Reserved at beginning of gl_pathv. */
+	int gl_flags;		/* Copy of flags parameter to glob. */
+	char **gl_pathv;	/* List of paths matching pattern. */
+	zend_stat_t **gl_statv;	/* Stat entries corresponding to gl_pathv */
+				/* Copy of errfunc parameter to glob. */
+	int (*gl_errfunc)(const char *, int);
+
+	/*
+	 * Alternate filesystem access methods for glob; replacement
+	 * versions of closedir(3), readdir(3), opendir(3), stat(2)
+	 * and lstat(2).
+	 */
+	void (*gl_closedir)(void *);
+	struct dirent *(*gl_readdir)(void *);
+	void *(*gl_opendir)(const char *);
+	int (*gl_lstat)(const char *, zend_stat_t *);
+	int (*gl_stat)(const char *, zend_stat_t *);
+} php_glob_t;
+
+#define	PHP_GLOB_APPEND	0x0001	/* Append to output from previous call. */
+#define	PHP_GLOB_DOOFFS	0x0002	/* Use gl_offs. */
+#define	PHP_GLOB_ERR	0x0004	/* Return on error. */
+#define	PHP_GLOB_MARK	0x0008	/* Append / to matching directories. */
+#define	PHP_GLOB_NOCHECK	0x0010	/* Return pattern itself if nothing matches. */
+#define	PHP_GLOB_NOSORT	0x0020	/* Don't sort. */
+#define	PHP_GLOB_NOESCAPE	0x1000	/* Disable backslash escaping. */
+
+#define	PHP_GLOB_NOSPACE	(-1)	/* Malloc call failed. */
+#define	PHP_GLOB_ABORTED	(-2)	/* Unignored error. */
+#define	PHP_GLOB_NOMATCH	(-3)	/* No match and PHP_GLOB_NOCHECK not set. */
+#define	PHP_GLOB_NOSYS	(-4)	/* Function not supported. */
+
+#define	PHP_GLOB_ALTDIRFUNC	0x0040	/* Use alternately specified directory funcs. */
+#define	PHP_GLOB_BRACE	0x0080	/* Expand braces ala csh. */
+#define	PHP_GLOB_MAGCHAR	0x0100	/* Pattern had globbing characters. */
+#define	PHP_GLOB_NOMAGIC	0x0200	/* PHP_GLOB_NOCHECK without magic chars (csh). */
+#define	PHP_GLOB_QUOTE	0x0400	/* Quote special chars with \. */
+#define	PHP_GLOB_TILDE	0x0800	/* Expand tilde names from the passwd file. */
+#define PHP_GLOB_LIMIT	0x2000	/* Limit pattern match output to ARG_MAX */
+#define	PHP_GLOB_KEEPSTAT	0x4000	/* Retain stat data for paths in gl_statv. */
```

Comment:
Nit: Do we care that this is aligned or not? In any case it probably should use spaces instead of tabs (unless it is to reduce diff with upstream)